generators/gateway/card_type_generator.rb

```diff
@@ -0,0 +1,82 @@
+require 'thor/group'
+
+class CardTypeGenerator < Thor::Group
+  include Thor::Actions
+
+  argument :card_type, type: :string, desc: "The name of the card type to add"
+  class_option :gateways, type: :array, desc: "List of gateway names to update (optional)", default: nil
+
+  def self.source_root
+    File.dirname(__FILE__)
+  end
+
+  def add_card_type_to_gateways
+    gateway_dir = "lib/active_merchant/billing/gateways"
+    test_dir = "test/unit/gateways"
+
+    gateways_to_update = if options[:gateways]
+      options[:gateways].map { |g| "#{gateway_dir}/#{g}.rb" }
+    else
+      Dir.glob("#{gateway_dir}/*.rb")
+    end
+
+    gateways_to_update.each do |gateway_path|
+      next unless File.exist?(gateway_path)
+
+      gateway_name = File.basename(gateway_path, ".rb")
+      test_path = "#{test_dir}/#{gateway_name}_test.rb"
+
+      if File.exist?(test_path)
+        update_gateway_file(gateway_path, card_type)
+        update_test_file(test_path, card_type)
+        say_status :update, "Added #{card_type} to #{gateway_name}", :green
+      else
+        say_status :skip, "Test file not found for #{gateway_name}", :yellow
+      end
+    end
+  end
+
+  private
+
+  def update_gateway_file(path, new_card_type)
+    content = File.read(path)
+    # This method updates the supported_cardtypes array in a gateway file
+    # It uses regex to find the line that defines supported_cardtypes, e.g:
+    #   self.supported_cardtypes = %i[visa mastercard amex]
+    # The regex captures 3 groups:
+    # $1 = "self.supported_cardtypes = %i["
+    # $2 = the current card types (e.g. "visa mastercard amex")
+    # $3 = "]"
+    updated_content = content.gsub(/(self\.supported_cardtypes\s*=\s*%i\[)(.*?)(\])/m) do
+      current_types = $2.strip
+      return if current_types.include?(new_card_type)
+      "#{$1}#{current_types.empty? ? new_card_type : "#{current_types} #{new_card_type}"}#{$3}"
+    end
+    File.write(path, updated_content)
+  end
+
+  def update_test_file(path, new_card_type)
```

Comment:
💬 Comment

This is the world we live in now, a proper prompt will compete with our best Regex:

prompt => 
`I want add a new type of card called "dava" to the list of supported types for the 3 gateways listed, I also want to update the corresponding unit tests adding "dava" to the list of cards that the test checks`

![Screenshot 2025-01-20 at 4 34 16 PM](https://github.com/user-attachments/assets/924c73d7-aacc-4bd9-995d-883cc286ab65)

Reply:
![Screenshot 2025-01-20 at 4 37 09 PM](https://github.com/user-attachments/assets/f22c6f94-fc0d-4063-a6a2-483704dfa444)

Reply:
![Screenshot 2025-01-20 at 4 37 42 PM](https://github.com/user-attachments/assets/329f0a6f-3e4d-49fe-a6b7-882490162665)

Reply:
Yes, it would be great to figure out how we can automate this kind of update using scripts or prompts like this